File: 品味Java的21種設計模式/dp_java/prototype/example8/ConcretePrototype2.java
package cn.javass.dp.prototype.example8;

/**
 * 克隆的具体实现对象
 */
public class ConcretePrototype2 implements Prototype {
	private String name;
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
	public Prototype clone() {
		ConcretePrototype2 prototype = new ConcretePrototype2();
		prototype.setName(this.name);
		return prototype;
	}
	
	public String toString(){
		return "Now in Prototype2，name="+name;
	}
}

File: 品味Java的21種設計模式/dp_cpp/prototype/example8/ConcretePrototype2.cpp
#include "ConcretePrototype2.h"

namespace cn {
namespace javass {
namespace dp {
namespace prototype {
namespace example8 {

std::string ConcretePrototype2::getName() {
    return name;
}

void ConcretePrototype2::setName(std::string name) {
    this->name = name;
}

Prototype *ConcretePrototype2::clone() {
    ConcretePrototype2 *prototype = new ConcretePrototype2();
    prototype->setName(this->name);
    return prototype;
}

std::string ConcretePrototype2::ToString() {
    return "Now in Prototype2，name=" + name;
}
}
}
}
}
}
